alt-tab-macos/ui/PreferencesPanel.swift

```diff
@@ -1,110 +1,211 @@
 import Cocoa
 
-class PreferencesPanel: NSPanel, NSTextViewDelegate {
-    var maxScreenUsage: NSTextView?
-    var windowPadding: NSTextView?
-    var cellPadding: NSTextView?
-    var cellBorderWidth: NSTextView?
-    var maxThumbnailsPerRow: NSTextView?
-    var iconSize: NSTextView?
-    var fontHeight: NSTextView?
-    var interItemPadding: NSTextView?
-    var tabKeyCode: NSTextView?
-    var windowDisplayDelay: NSTextView?
+class PreferencesPanel: NSPanel {
+    // ui: base layout
+    let panelWidth = CGFloat(400)
+    let panelHeight = CGFloat(400) // gets auto adjusted to content height
+    let panelPadding = CGFloat(40)
+    let panelWidthToLabelRatio = CGFloat(0.5)
+
+    // ui: preferences elements
+    var maxScreenUsage: NSSlider?
+    var maxThumbnailsPerRow: NSTextField?
+    var iconSize: NSSlider?
+    var fontHeight: NSSlider?
+    var tabKeyCode: NSTextField?
+    var windowDisplayDelay: NSTextField?
     var metaKey: NSPopUpButton?
     var theme: NSPopUpButton?
     var showOnScreen: NSPopUpButton?
-    var inputsMap = [NSTextView: String]()
+
+    var invisibleTextField: NSTextField? // default firstResponder and used for triggering of focus loose
```

Comment:
I could find a other way to unfocus an input and to prevent the NSPanel from focusing the first input element when its opened.

Ive tried making other elements become firstResponder but only succeeded with input elements. I was not ready to invest more time on it and went for the invisibleTextField until a better way is found (more NS... knowledge). Do you know maybe how to do this? Iam probably doing something wrong here.

The programmatic unfocus is required when:
1. use case
 - click input element
 - click a dropdown
 - result: input element is still focused and will not trigger `controlDidEndEditing` until some other input is focused
2. use case
 - click input element
 - click restart button
 - result: `controlDidEndEditing` will again be not triggered and that way the user change not saved

Reply:
Maybe MVC could be the solution here but I did not look into it yet. Only saw Controllers being used together with Views. I hope there is some binding and automatic changeDetection similar to Angular.

Reply:
> input element is still focused and will not trigger controlDidEndEditing until some other input is focused

Oh, that's why? I'm surprised by this behavior. I would have to reproduce and also read online what long time mac developers have been writing about it. I'm assuming it's either a bad setup on our part, of it's a silly Cocoa design limitation that must have known work-arounds at this point